push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/configuration/SenderConfigurationProvider.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.configuration;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import org.jboss.aerogear.unifiedpush.api.VariantType;
+import org.jboss.aerogear.unifiedpush.message.sender.SenderType;
+
+/**
+ * Loads and stores configuration for specific Push Networks.
+ *
+ * UPS provides sensible defaults specific to each Push Network and its limitations.
+ *
+ * Configuration can be changed by overriding system properties, e.g. via as VM args.
+ *
+ * Example: In order to change the configuration of {@link SenderConfiguration#batchSize()} for APNs (iOS) service,
+ * one can provide following system property: <tt>-Daerogear.ios.batchSize=12000</tt>.
+ *
+ * The name of the network (<tt>aerogear.&lt;network&gt;.&lt;property&gt;</tt>) is specified by {@link VariantType#getTypeName()}.
+ *
+ * Look at {@link SenderConfiguration} for more details about available configurations.
+ *
+ * @see SenderConfiguration
+ */
+public class SenderConfigurationProvider {
+
+    @Produces @ApplicationScoped @SenderType(VariantType.ANDROID)
+    public SenderConfiguration produceAndroidConfiguration() {
+        return loadConfigurationFor(VariantType.ANDROID, new SenderConfiguration(10, 1000));
+    }
+
+    @Produces @ApplicationScoped @SenderType(VariantType.ADM)
+    public SenderConfiguration produceAdmConfiguration() {
+        return loadConfigurationFor(VariantType.ADM, new SenderConfiguration(10, 1000));
+    }
+
+    @Produces @ApplicationScoped @SenderType(VariantType.IOS)
+    public SenderConfiguration produceIosConfiguration() {
+        return loadConfigurationFor(VariantType.IOS, new SenderConfiguration(3, 10000));
```

Comment:
That is the number of batches that will be loaded in one transaction. The transaction may not end on time, because the JMS queue will block it until it has enough memory to receive it.

So the number of batches loaded in one transaction is a trade-off between performance and need for repeating the work that was already done.

Details in:
https://docs.google.com/a/fryc.eu/drawings/d/13IsJWPSJNYXtst-UVxQYmzH36C_EXQMYYr_jcu7nFmE/edit?usp=drive_web